src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/ExclusiveJobSchedulerDAO.java

```diff
@@ -71,7 +71,7 @@ int scheduleJob(
     @Query(
         "local deletedFromReadyQueue = redis.call('ZREM', $readyQueue$, $job$)\n" +
         "local deletedFromRunningQueue = redis.call('ZREM', $runningQueue$, $job$)\n" +
-        "if not deletedFromReadyQueue and not deletedFromRunningQueue then\n" +
+        "if deletedFromReadyQueue == 0 and deletedFromRunningQueue == 0 then\n" +
```

Comment:
we never really tested this in rdbi. when i added test cases they broke for me because the 0 was not being considered null which is what the "not" check does. i have redis 2.6.16 running locally.  so unless 2.8 behaves differently this was a bug.